### ISSUE TITLE
Support pug templates in TemplateLiterals

### ISSUE THREAD

**Issue by ezhlobo** [pugjs/pug-lint#151]
Babel plugin [babel-plugin-transform-react-pug](https://github.com/pugjs/babel-plugin-transform-react-pug) becomes more popular and I want to set up linting of its templates as well.

We need to allow custom parsers for this linter or something else. So we can parse `.js` files, extract pug templates from there and lint only pug.

Example of javascript file:

```js
import React from 'react'

const MyComponent = props => pug`
  div
    p= props.text
`

export default MyComponent
```

What should be linted in that file:

```pug
  div
    p= props.text
```
